### PR TITLE
docs(openapi): add readme of OpenAPI section

### PR DESCRIPTION
**Pull request by liubog2008**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes #

**Special notes for your reviewer**:

/cc @ddysher 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -854,7 +854,96 @@ Operators: []definition.Operator{
...
+if err := encoder.Encode(swagger); err != nil {
+    panic(err)
+}
+```
+
+You can also serve the documents in an OpenAPI endpoint, e.g. /v2/openapi.
+`NOTICE: Don't add openapi descriptor into the descriptor passed to the builder.`
+
+```
+var openapi = definition.Descriptor{
```
Comment (ddysher) [inline README.md]: let's make this a plugin?

File: README.md [hunk 2]
```diff
@@ -854,7 +854,96 @@ Operators: []definition.Operator{
...
+```
+// +nirvana:openapi=true
+package api
+```
+
+And run cmd to generate the function `GetOpenAPIDefinitions`
+
+```
+openapi-gen \
+  -i /go/package/to/your/types \
```
Comment (ddysher) [inline README.md]: what if i have a lot of types to generate?

File: README.md [hunk 3]
```diff
@@ -854,7 +854,96 @@ Operators: []definition.Operator{
 
 ### OpenAPI
 
-TODO
+Nirvana can generate OpenAPI 2.0 document from code simply.
+
+In the example, swagger will be generated by builder of OpenAPI spec.
```
Comment (ddysher) [inline README.md]: we already have example above, i think this should be a thorough guide.  what it is, how it works, etc

File: README.md (hunk 1 above)
Comment (liubog2008) [reply README.md]: LGTM
I'll file a issue to record

File: README.md (hunk 3 above)
Comment (liubog2008) [reply README.md]: ?!!!

Comment (liubog2008) [reply README.md]: What I need to explain? How to generate code?

Comment (ddysher) [reply README.md]: let's discuss this offline tmr

**Comment by ddysher** [comment]
@liubog2008 can u update mail address?

**Comment by liubog2008** [comment]
@ddysher updated

**Pull request merged**